runtime/common/src/xcm_impl.rs

```diff
@@ -0,0 +1,491 @@
+	fn from(currency_id: CurrencyId<R>) -> Self {
+		match currency_id {
+			// For now and until Xtokens is adapted to handle 0.9.16 version we use
+			// the old anchoring here
+			// This is not a problem in either cases, since the view of the destination
+			// chain does not change
+			// TODO! change this to NewAnchoringSelfReserve once xtokens is adapted for it
+			CurrencyId::<R>::SelfReserve(_) => {
+				let multi: MultiLocation = OldAnchoringSelfReserve::<R>::get();
+				Some(multi)
+			},
+			CurrencyId::<R>::ParachainReserve(multi) => Some(*multi),
+		}
+	}
+}
+
+// How to convert from CurrencyId to MultiLocation: for orml convert sp_runtime Convert
+// trait
+pub struct CurrencyIdMultiLocationConvert<R: BaseRuntimeRequirements>(PhantomData<R>);
+impl<R: BaseRuntimeRequirements> spConvert<CurrencyId<R>, Option<MultiLocation>>
+	for CurrencyIdMultiLocationConvert<R>
+{
+	fn convert(currency: CurrencyId<R>) -> Option<MultiLocation> {
+		currency.into()
+	}
+}
+
+impl<R: BaseRuntimeRequirements> spConvert<MultiLocation, Option<CurrencyId<R>>>
+	for CurrencyIdMultiLocationConvert<R>
+{
+	fn convert(multi: MultiLocation) -> Option<CurrencyId<R>> {
+		match multi {
+			a if (a == OldAnchoringSelfReserve::<R>::get()) |
+				(a == NewAnchoringSelfReserve::<R>::get()) =>
+				Some(CurrencyId::<R>::SelfReserve(PhantomData::default())),
+			_ => Some(CurrencyId::<R>::ParachainReserve(Box::new(multi))),
+		}
+	}
+}
+
+/// Converter struct implementing `AssetIdConversion` converting a numeric asset ID
+/// (must be `TryFrom/TryInto<u128>`) into a MultiLocation Value and Viceversa through
+/// an intermediate generic type AssetType.
+/// The trait bounds enforce is that the AssetTypeGetter trait is also implemented
+pub struct AssetIdMuliLocationConvert<R>(PhantomData<R>);
+impl<R: ParaRuntimeRequirements> xcmConvert<MultiLocation, AssetId>
+	for AssetIdMuliLocationConvert<R>
+where
+	R: pallet_asset_manager::Config<ForeignAssetType = CurrencyId<R>>,
+{
+	fn convert_ref(multi: impl Borrow<MultiLocation>) -> Result<AssetId, ()> {
+		if let Some(currency_id) = <CurrencyIdMultiLocationConvert<R> as spConvert<
+			MultiLocation,
+			Option<CurrencyId<R>>,
+		>>::convert(multi.borrow().clone())
+		{
+			if let Some(asset_id) =
+				<AssetManager<R> as AssetTypeGetter<AssetId, CurrencyId<R>>>::get_asset_id(
+					currency_id,
+				) {
+				Ok(asset_id)
+			} else {
+				Err(())
+			}
+		} else {
+			Err(())
+		}
+	}
+
+	fn reverse_ref(asset_id: impl Borrow<AssetId>) -> Result<MultiLocation, ()> {
+		if let Some(currency_id) =
+			<AssetManager<R> as AssetTypeGetter<AssetId, CurrencyId<R>>>::get_asset_type(
+				*asset_id.borrow(),
+			) {
+			if let Some(multi) = <CurrencyIdMultiLocationConvert<R> as spConvert<
+				CurrencyId<R>,
+				Option<MultiLocation>,
+			>>::convert(currency_id)
+			{
+				Ok(multi)
+			} else {
+				Err(())
+			}
+		} else {
+			Err(())
+		}
+	}
+}
+
+// implement different types of converters:
```

Comment:
We can delete these comments if not needed, unless it's worth keeping it in your opinion

Reply:
had been deleted.